cmd/podman/kube/play.go

```diff
@@ -253,19 +253,34 @@ func play(cmd *cobra.Command, args []string) error {
 	}
 
 	if playOptions.Down {
-		return teardown(reader, entities.PlayKubeDownOptions{Force: playOptions.Force})
+		return teardown(reader, entities.PlayKubeDownOptions{Force: playOptions.Force}, false)
 	}
 
 	if playOptions.Replace {
-		if err := teardown(reader, entities.PlayKubeDownOptions{Force: playOptions.Force}); err != nil && !errorhandling.Contains(err, define.ErrNoSuchPod) {
+		if err := teardown(reader, entities.PlayKubeDownOptions{Force: playOptions.Force}, false); err != nil && !errorhandling.Contains(err, define.ErrNoSuchPod) {
 			return err
 		}
 		if _, err := reader.Seek(0, 0); err != nil {
 			return err
 		}
 	}
 
-	return kubeplay(reader)
+	if err := kubeplay(reader); err != nil {
+		// teardown any containers, pods, and volumes that might have created before we hit the error
+		teardownReader, trErr := readerFromArg(args[0])
+		if trErr != nil {
+			return trErr
+		}
+		defer func() {
```

Comment:
The defer here isn't needed since we return right after.  Looks like there is only one place where we need to cleanup and that's here.

Reply:
reverted the change to remove the defer